// @ionic/angular/esm5/providers/gesture-controller.js
import * as tslib_1 from "tslib";
import { Injectable, NgZone } from '@angular/core';
import { createGesture } from '@ionic/core';
import * as i0 from "@angular/core";
import * as ɵngcc0 from '@angular/core';
var GestureController = /** @class */ (function () {
    function GestureController(zone) {
        this.zone = zone;
    }
    /**
     * Create a new gesture
     */
    GestureController.prototype.create = function (opts, runInsideAngularZone) {
        var _this = this;
        if (runInsideAngularZone === void 0) { runInsideAngularZone = false; }
        if (runInsideAngularZone) {
            Object.getOwnPropertyNames(opts).forEach(function (key) {
                if (typeof opts[key] === 'function') {
                    var fn_1 = opts[key];
                    opts[key] = function () {
                        var props = [];
                        for (var _i = 0; _i < arguments.length; _i++) {
                            props[_i] = arguments[_i];
                        }
                        return _this.zone.run(function () { return fn_1.apply(void 0, tslib_1.__spread(props)); });
                    };
                }
            });
        }
        return createGesture(opts);
    };
    GestureController.ctorParameters = function () { return [
        { type: NgZone }
    ]; };
    GestureController.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function GestureController_Factory() { return new GestureController(i0.ɵɵinject(i0.NgZone)); }, token: GestureController, providedIn: "root" });
GestureController.ɵfac = function GestureController_Factory(t) { return new (t || GestureController)(ɵngcc0.ɵɵinject(ɵngcc0.NgZone)); };
GestureController.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: GestureController, factory: function (t) { return GestureController.ɵfac(t); }, providedIn: 'root' });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GestureController, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: ɵngcc0.NgZone }]; }, null); })();
    return GestureController;
}());
export { GestureController };
